src/CMake/FindVisItPython.cmake

```diff
@@ -457,115 +451,76 @@ IF(PYTHONLIBS_FOUND AND NOT VISIT_PYTHON_SKIP_INSTALL)
-            INSTALL(DIRECTORY ${PyPth}/
-                DESTINATION ${VISIT_INSTALLED_VERSION_LIB}/python/include
-                FILE_PERMISSIONS OWNER_READ OWNER_WRITE
-                                 GROUP_READ GROUP_WRITE
-                                 WORLD_READ
-                DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
-                                      GROUP_READ GROUP_WRITE GROUP_EXECUTE
-                                      WORLD_READ             WORLD_EXECUTE
-                FILES_MATCHING PATTERN "*.h"
-                PATTERN ".svn" EXCLUDE
-            )
-            INSTALL(DIRECTORY ${VISIT_PYTHON_DIR}/lib
+        if(VISIT_HEADERS_SKIP_INSTALL)
+            message(STATUS "Skipping python headers installation")
+        else()
+            # KSB
+            #  WIN32  PYTHON_INCLUDE_PATH is
+            #    'path-to-python/include'
+            #  Non-WIN32 PYTHON_INCLUDE_PATH is: 
+            #    'path-to-python/include/python<vermaj>.<vermin>m'
+            #  So Non-WIN32 needs extra 'include' appended to DESTINATION
+                
+            set(pyIncDest ${VISIT_INSTALLED_VERSION_INCLUDE}/python)
+            if(NOT WIN32)
+                string(APPEND pyIncDest "/include")
+            endif()
+            install(DIRECTORY ${PYTHON_INCLUDE_PATH}
+                DESTINATION ${pyIncDest}
+                FILE_PERMISSIONS ${filePerms}
+                DIRECTORY_PERMISSIONS ${dirPerms})
+        endif()
+        #
+        # CDH:
+        # We also need to install the headers into lib/python dir
+        # if we want to be able to install python modules into a visit
+        # install.
+        #
+        set(pyIncDest ${VISIT_INSTALLED_VERSION_LIB}/python)
+        if(NOT WIN32)
+            string(APPEND pyIncDest "/include")
+        endif()
+        install(DIRECTORY ${PYTHON_INCLUDE_PATH}
+            DESTINATION ${pyIncDest}
+            FILE_PERMISSIONS ${filePerms}
+            DIRECTORY_PERMISSIONS ${dirPerms})
+
+        if(WIN32)
+            # install the import libraries
+            install(DIRECTORY ${VISIT_PYTHON_DIR}/lib
                 DESTINATION ${VISIT_INSTALLED_VERSION_LIB}/python
-                FILE_PERMISSIONS OWNER_READ OWNER_WRITE
-                                 GROUP_READ GROUP_WRITE
-                                 WORLD_READ
-                DIRECTORY_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
-                                      GROUP_READ GROUP_WRITE GROUP_EXECUTE
-                                      WORLD_READ             WORLD_EXECUTE
-                PATTERN ".svn"   EXCLUDE
-            )
-            unset(PyPth)
-        ENDIF (NOT WIN32)
-    ENDIF((NOT ${PYTHON_DIR} STREQUAL "/usr"))
-ENDIF(PYTHONLIBS_FOUND AND NOT VISIT_PYTHON_SKIP_INSTALL)
+                FILE_PERMISSIONS ${filePerms}
+                DIRECTORY_PERMISSIONS ${dirPerms})
+        endif()
+        unset(filePerms)
+        unset(dirPerms)
+    endif()
+endif()
```

Comment:
Having never been here before, to me this CMake logic to find python seems hideously complex. Are we trying to be too adaptable to whatever the local python installation might look like? Any ideas for simplification?

I don't think you really need to answer in this PR but...wow...that is a lot of CMake code :wink:

Reply:
Well, the logic to Find python ends on line 299. The rest is logic for installing, and functions for adding python modules.  You would have to query @cyrush regarding the complexity of the find logic.

Reply:
@cyrush can you comment on what is driving complexity in this `FindVisItPytyhon.cmake` logic. Doesn't BLT have stuff to handle most/all of this?